python/cudf/cudf/core/column/column.py

```diff
@@ -1380,7 +1378,7 @@ def _return_sentinel_column():
         # values of `self`:
         order = arange(len(self))
         order = order.take(left_gather_map, check_bounds=False).argsort()
-        codes = codes.take(order)
+        codes = codes.take(order, check_bounds=False)
```

Comment:
Reviewing uses of `take` to avoid bounds-checking would be a good thing anywhere that cudf constructs the indices (and they're known to be safe) rather than taking user input.

Reply:
#13456